File: examples/cpp/my-app/src/services/NotificationService.cpp
#include "services/NotificationService.h"
#include "utils/Logger.h"
#include <iostream>

namespace services {

void NotificationService::sendNotification(int userId, const std::string& message) {
    utils::Logger::getInstance().info("Sending notification to user " + std::to_string(userId) + ": " + message);
    std::cout << "Notification for user " << userId << ": " << message << std::endl;
}

void NotificationService::sendEmailNotification(const std::string& email, const std::string& subject, const std::string& body) {
    utils::Logger::getInstance().info("Sending email to " + email + " - Subject: " + subject);
    std::cout << "Email to " << email << "\nSubject: " << subject << "\n" << body << std::endl;
}

std::vector<std::string> NotificationService::getNotifications(int userId) {
    // In a real app, this would fetch from database
    return {};
}

} // namespace services

File: examples/python/my-app/src/services/notification_service.py
"""Notification service for sending notifications."""

from models.task import Task
from utils.logger import Logger

class NotificationService:
    """Service for sending notifications."""
    
    def __init__(self):
        self._logger = Logger.get_instance()
    
    def notify_task_created(self, task: Task) -> None:
        """Send notification when task is created."""
        self._logger.info(f"Sending notification: Task created - {task.title}")
        # In real app, would send email/push notification
    
    def notify_task_completed(self, task: Task) -> None:
        """Send notification when task is completed."""
        self._logger.info(f"Sending notification: Task completed - {task.title}")
    
    def notify_task_assigned(self, task: Task, assignee_id: int) -> None:
        """Send notification when task is assigned."""
        self._logger.info(f"Sending notification: Task assigned to user {assignee_id}")
    
    def notify_user_created(self, username: str, email: str) -> None:
        """Send notification when user is created."""
        self._logger.info(f"Sending notification: User created - {username}")
